pkg/serialization/codec/jam/decode.go

```diff
@@ -521,23 +522,30 @@ func (br *byteReader) decodeBits(dstv reflect.Value) error {
 	return nil
 }
 
-func (br *byteReader) decodeBitsFixedLength(v *BitSequence, bytesLength uint) (err error) {
-	if bytesLength > math.MaxUint32 {
-		return ErrExceedingByteArrayLimit
+func (br *byteReader) decodeBitsFixedLength(v *BitSequence, bitLength uint) error {
+	if bitLength == 0 {
+		return nil
 	}
-	bb := make([]byte, bytesLength)
-	if _, err = br.Reader.Read(bb); err != nil {
-		return err
+
+	numerator, ok := safemath.Add(bitLength, 7)
+	if !ok {
+		return safemath.ErrOverflow
 	}
-	if bytesLength == 0 {
-		return nil
+	byteLength := numerator / 8
+	if byteLength > math.MaxUint32 {
+		return ErrExceedingByteArrayLimit
+	}
+
+	bb := make([]byte, byteLength)
+	if byteLength > 0 {
+		if _, err := io.ReadFull(br.Reader, bb); err != nil {
+			return err
+		}
 	}
```

Comment:
At this point it will always be `>0`.
```suggestion
	  if _, err := io.ReadFull(br.Reader, bb); err != nil {
      return err
  }
```